packages/openid4vc/src/openid4vc-holder/OpenId4VciHolderService.ts

```diff
@@ -0,0 +1,719 @@
+import type {
+  OpenId4VciCredentialOfferPayload,
+  OpenId4VciCredentialSupported,
+  OpenId4VciCredentialSupportedWithId,
+  OpenId4VciIssuerMetadata,
+} from '../shared'
+import type {
+  AgentContext,
+  JwaSignatureAlgorithm,
+  W3cVerifiableCredential,
+  Key,
+  JwkJson,
+  SdJwtVc,
+} from '@aries-framework/core'
+import type {
+  AccessTokenResponse,
+  CredentialResponse,
+  Jwt,
+  OpenIDResponse,
+  PushedAuthorizationResponse,
+  AuthorizationDetails,
+  AuthorizationDetailsJwtVcJson,
+} from '@sphereon/oid4vci-common'
+
+import {
+  SdJwtVcApi,
+  getJwkFromJson,
+  DidsApi,
+  AriesFrameworkError,
+  Hasher,
+  InjectionSymbols,
+  JsonEncoder,
+  JwsService,
+  Logger,
+  SignatureSuiteRegistry,
+  TypedArrayEncoder,
+  W3cCredentialService,
+  W3cJsonLdVerifiableCredential,
+  W3cJwtVerifiableCredential,
+  getJwkClassFromJwaSignatureAlgorithm,
+  getJwkFromKey,
+  getKeyFromVerificationMethod,
+  getSupportedVerificationMethodTypesFromKeyType,
+  inject,
+  injectable,
+  parseDid,
+} from '@aries-framework/core'
+import {
+  AccessTokenClient,
+  CredentialRequestClientBuilder,
+  ProofOfPossessionBuilder,
+  formPost,
+  OpenID4VCIClient,
+} from '@sphereon/oid4vci-client'
+import { CodeChallengeMethod, ResponseType, convertJsonToURI, JsonURIMode } from '@sphereon/oid4vci-common'
+
+import { OpenId4VciCredentialFormatProfile } from '../shared'
+import {
+  getTypesFromCredentialSupported,
+  handleAuthorizationDetails,
+  getOfferedCredentials,
+} from '../shared/issuerMetadataUtils'
+import { getSupportedJwaSignatureAlgorithms } from '../shared/utils'
+
+import {
+  type OpenId4VciAuthCodeFlowOptions,
```

Comment:
Shouldn't we use `import type { ...`? Or will this work always now? Vaguely remember that this caused some issues with me before, but it might've been resolved now.

Reply:
I think the issue we had recently was to do `export type * from` which is TS 5+. But I usually avoid this as well jsut to prevent TS version issues. Will update